packages/EasyWebhook/docs/middleware.md

```diff
@@ -0,0 +1,217 @@
+
+Note that the dispatcher will retrieve the webhook from the store and trigger the processing of the middleware stack
+again. The webhook is flagged as configured due to being retrieved from the store, so configure once middleware will not
+operate on the webhook.
+
+### `BodyFormatterMiddleware`
+
+This middleware formats the body of the webhook HTTP request as a JSON string and sets the `Content-Type` header of the
+HTTP request to be `application/json`.
+
+*This is configure once middleware.*
+
+### `EventHeaderMiddleware`
+
+This middleware sets the `X-Webhook-Event` header of the webhook HTTP request to the webhook's `$event` property (if it
+exists).
+
+*This is configure once middleware.*
+
+### `EventsMiddleware`
+
+This middleware dispatches an event depending on the outcome of the webhook HTTP request. See see [Events](events.md))
+for more information.
+
+### `IdHeaderMiddleware`
+
+This middleware sets the `X-Webhook-Id` header of the webhook HTTP request. It uses the webhook's `$id` property (if it
+exists) or generates a new ID.
+
+*This is configure once middleware.*
+
+### `LockMiddleware`
+
+This middleware locks the Webhook object at the start of middleware processing and unlocks it at the end.
+
+### `MethodMiddleware`
+
+This middleware sets the HTTP method for sending the webhook HTTP request.
+
+*This is configure once middleware.*
+
+### `RerunMiddleware`
+
+If the webhook is in a final state (i.e. success or failed), this middleware checks if rerun is allowed for the webhook,
+and if so, resets its `$currentAttempt` and its `$status` to `pending`, which allows the webhook to be rerun.
+
+### `ResetStoreMiddleware`
+
+This middleware resets the webhook store and webhook results store. See [Stores](stores.md) for more information.
+
+Note that the stores must implement `EonX\EasyWebhook\Interfaces\Stores\ResetStoreInterface`. Of the stores provided by
+the EasyWebhook package, only the array stores support reset.
+
+### `SendAfterMiddleware`
+
+This middleware checks the `$sendAfter` property of the webhook, and if it exists and is after the current date and
+time, it simply stores the webhook in the configured store, and does not proceed with any further middleware processing.
+You can initiate sending of delayed webhooks via the console command `easy-webhooks:send-due-webhooks`. See
+[Console](console.md) for more information.
+
+Note that the webhook store must implement `EonX\EasyWebhook\Interfaces\Stores\SendAfterStoreInterface` in order to be
+able to find due webhooks. Of the stores provided by the EasyWebhook package, only the Doctrine DBAL webhook store
+supports finding due webhooks.
+
+### `SendWebhookMiddleware`
+
+This middleware sends the webhook HTTP request and returns the result in a [WebhookResult](webhook-result.md) object.
+
+### `SignatureHeaderMiddleware`
+
+This middleware sets the `X-Webhook-Signature` header of the webhook HTTP request. The header contains a SHA-256 HMAC
```

Comment:
Here we should say that the signature logic can be customised by creating a new implementation of the signature service and telling the package to use it.
And also that the header name is configurable.